BedrockServer.cpp

```diff
@@ -1007,12 +1006,48 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
             break;
         }
 
-        // We're about to retry, decrement the retry count.
-        --retry;
+        // If we're shutting down, or have set a specific max retries, we just try several times in a row and then move the command to the blocking queue.
+        int maxRetries = _maxConflictRetries.load();
+        if (maxRetries || _shutdownState.load() != RUNNING) {
+            if (command->processCount > maxRetries) {
+                SINFO("Max retries (" << maxRetries << ") hit in worker, sending '" << command->request.methodLine << "' to blocking queue with size " << _blockingCommandQueue.size());
+                _blockingCommandQueue.push(move(command));
+                return;
+            }
+        } else {
+            // If we're not shutting down, see how long we want to wait until we'll try this command again.
+            size_t millisecondsToWait = 0;
+            switch (command->processCount) {
+                case 1:
+                    millisecondsToWait = 10;
+                    break;
+                case 2:
+                    millisecondsToWait = 25;
+                    break;
+                case 3:
+                    millisecondsToWait = 50;
+                    break;
+                case 4:
+                    millisecondsToWait = 100;
+                    break;
+                case 5:
+                    millisecondsToWait = 250;
+                    break;
+                default:
+                    millisecondsToWait = 500;
```

Comment:
If we have 2 commands running for 1 second which conflict all the time (say, ReimburseReport), will they simply never complete?

Reply:
One of them will complete on the first pass one the other will complete on the second pass. When there's a conflict, one commit succeeds and the other is repeated. They don't both fail.